code/__HELPERS/areas.dm

```diff
@@ -141,6 +183,17 @@ GLOBAL_LIST_INIT(typecache_powerfailure_safe_areas, typecacheof(/area/station/en
 	SEND_GLOBAL_SIGNAL(COMSIG_AREA_CREATED, newA, oldA, creator)
 	to_chat(creator, span_notice("You have created a new area, named [newA.name]. It is now weather proof, and constructing an APC will allow it to be powered."))
 	creator.log_message("created a new area: [AREACOORD(creator)] (previously \"[oldA.name]\")", LOG_GAME)
+
+	/**
+	 * there are no more turfs left in this area. Time to clean up
+	 * we could have used has_contained_turfs() to determine if the old area has any turfs remaining but insead
+	 * we use cannonize_contained_turfs() because it saves memory as an added bonus
+	 */
+	oldA.cannonize_contained_turfs()
+	if(!length(oldA.contained_turfs))
+		message_admins("Area [oldA.name] has been deleted!.")
+		qdel(oldA)
```

Comment:
this doesn't "save memory" it potentially dumps cpu usage into this proc.
uncontained turfs will slowly drop away, you are doing the job of `/datum/controller/subsystem/area_contents`. Don't.

Reply:
True but that creates a problem for us. Since `turfs_to_uncontain` gets cleared over time the proc `has_contained_turfs()` won't be able to determine over time if we need to delete the area or not.
https://github.com/tgstation/tgstation/blob/e7a5cfb3a8d219b9a6ae6330f5931491194f1510/code/game/area/areas.dm#L250-#L252
It will always return true when `turfs_to_uncontain()` becomes 0 when it gets cleared by the sub system and thus our empty area will exist forever. 

Unless we program the subsystem to delete emty areas this is nessassry

Reply:
This also creates problems for admins who want to jump to areas. When the area becomes empty it still gets listed and they get the error "No where to jump"

Reply:
Never mind I should have read the code first you were right.  I'll keep the above comments just for fun.  But yup I will change it